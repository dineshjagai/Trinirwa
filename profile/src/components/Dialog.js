import React from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import axios from 'axios';

export default function DialogBox(props) {
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
  const addUrl = '/profile/interest/'+props.id;
  let input = document.getElementById('interest').value;
  input = input+"";
    axios({
        method: 'POST',
        url: addUrl,
        data: {
            interest: input,
        }
    });
  
    setOpen(false);
    
  };

  return (
    <div>
      <Button  color="primary" onClick={handleClickOpen}>
        Add interest
      </Button>
      <Dialog open={open} onClose={handleClickOpen} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title">Add interest</DialogTitle>
        <DialogContent>
          <TextField
            autoFocus
            margin="dense"
            id="interest"
            type="string"
            fullWidth
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Cancel
          </Button>
          <Button onClick={handleClose} color="primary">
            Add
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
