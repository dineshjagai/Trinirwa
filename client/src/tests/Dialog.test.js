import React from 'react';
import renderer from 'react-test-renderer';
import Dialog from '../components/Dialog';

test('Test Dialog snapshot', () => {
  const component = renderer.create(<Dialog />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
