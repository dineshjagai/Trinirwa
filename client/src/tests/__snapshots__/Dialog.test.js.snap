// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dialog snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add interest
    </span>
  </button>
</div>
`;
